### ISSUE TITLE
Error validating token

### ISSUE THREAD

**Issue by jclif** [lynndylanhurley/devise_token_auth#348]
I am using j-toker to manage authentication, but when I refresh the page and it tries to validate the token, I get the following output in my server log:

```
Started GET "/api/v1/auth/validate_token" for 127.0.0.1 at 2015-08-19 21:41:34 -0400
Processing by DeviseTokenAuth::TokenValidationsController#validate_token as */*
Completed 500 Internal Server Error in 1ms (ActiveRecord: 0.0ms)

NoMethodError (undefined method `name' for nil:NilClass):
  devise (3.5.1) app/controllers/devise_controller.rb:38:in `resource_name'
  devise_token_auth (0.1.34) app/controllers/devise_token_auth/application_controller.rb:10:in `resource_class'
  devise_token_auth (0.1.34) app/controllers/devise_token_auth/concerns/set_user_by_token.rb:19:in `set_user_by_token'
```

Has anyone experienced this issue before?

**Comment by yivo** [comment]
Please try [this](https://github.com/lynndylanhurley/devise_token_auth/pull/355)

**Comment by booleanbetrayal** [comment]
should be fixed via #355

**Comment by ponyesteves** [comment]
Thanks! I've a similar problem using namespaces. Besides I have similar issue when authentication with facebook. Solved it same way (file app/controllers/devise_token_auth/omniauth_callbacks_controller.rb) 
    
    def redirect_callbacks
      devise_mapping = request.env['omniauth.params']['resource_class'].underscore.gsub('/','_').to_sym
      redirect_route = "/#{Devise.mappings[devise_mapping].as_json["path"]}/#{params[:provider]}/callback"

This was the error message

![image](https://cloud.githubusercontent.com/assets/4960589/9802024/818db4b0-57ed-11e5-991e-ead421b0dd82.png)
